### PR TITLE
Add timeout to build logging

### PR DESCRIPTION
**Pull request by meffij**

### CODE REVIEW

File: compiler/quilt/tools/command.py [hunk 1]
```diff
@@ -510,7 +511,7 @@ def _log(team, **kwargs):
     if cfg.get('disable_analytics'):
         return
 
-    session = _get_session(team)
+    session = _get_session(team, timeout=LOG_TIMEOUT)
```
Comment (dimaryaz) [inline compiler/quilt/tools/command.py]: Need to catch the exception. Move this into the try/except?

File: compiler/quilt/tools/command.py [hunk 2]
```diff
@@ -510,12 +511,12 @@ def _log(team, **kwargs):
     if cfg.get('disable_analytics'):
         return
 
-    session = _get_session(team)
+    try:
+        session = _get_session(team, timeout=LOG_TIMEOUT)
```
Comment (dimaryaz) [inline compiler/quilt/tools/command.py]: You also need to modify the `finally` block to check if `session` is set. Anyways, make sure to actually test the timeouts - just set it to 0.001 or something.

Comment (meffij) [reply compiler/quilt/tools/command.py]: Tested and works

File: compiler/quilt/tools/command.py [hunk 3]
```diff
@@ -212,9 +212,10 @@ def config():
     global _registry_url
     _registry_url = None
 
-def _update_auth(team, refresh_token):
+def _update_auth(team, refresh_token, timeout=None):
     response = requests.post("%s/api/token" % get_registry_url(team), data=dict(
-        refresh_token=refresh_token
+        refresh_token=refresh_token,
+        timeout=timeout
```
Comment (dimaryaz) [inline compiler/quilt/tools/command.py]: Wait a minute... You're passing it as a POST parameter. There's no way that can work.

**Pull request merged**